### PR TITLE
Added devfile registry docs

### PR DESCRIPTION
**Pull request by jc-berger**
Fixes [#191 Define onboarding requirement](https://github.com/devfile/api/issues/191)

Taking content from John's devfile-registry docs repo: https://github.com/johnmcollier/registry-docs 

I've left comments throughout the docs, asking questions to help direct peer reviewers' attention. 

Thanks! 

@johnmcollier or @GeekArthur would either of you take a look please to see how the devfile registry docs are looking?

### CODE REVIEW

File: docs/modules/user-guide/nav.adoc
Comment (jc-berger) [inline docs/modules/user-guide/nav.adoc:2]: Does its placement in the TOC look good? I thought getting started with the devfile registry would be the first step, hence, putting it right after the overview page.

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: I wanted to end on a statement targeted towards the users. So I said users can use devfile stacks as a template to get started with devfiles. That's correct, right?

Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: As for the links, once we get all the devfile registry docs included in our docs repo, they can be xrefs (because they'll be linking to our own docs). For now, they'll have to be outside webpages that we're linking to.

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:17]: The original first bullet point was: 
`Hosts metadata about the stacks and samples in the registry for tools to consume (via the index.json file).`

I found that hard to digest. Is this revision good and correct?

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:27]: Any opinion on how many links we have in this doc? I personally think we have too many. I think when there's this many links, it becomes _very easy_ for users to leave our page and fall down a rabbit hole. I wanted a second opinion. If anyone agrees and has an opinion on which links to eliminate, I'm all ears, thanks!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:15]: The original version was, 
`The server (and resulting container image) that does the following`

Could we say, `the server and *its* container do the following` 

I don't want to change meaning...also, if it's put in a parentheses, maybe it's not pertinent enough to include?

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:5]:
```suggestion
A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che, and the OpenShift Developer Console. Therefore, users can access devfiles through the devfile registry. Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification and devfile stacks are stored as OCI-artifacts in the registry.
```
IBM SG: "Use a serial comma (comma before and or or for three or more items)."

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]:
```suggestion
Each devfile stack corresponds to a specific runtime or framework such as NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources such as a `devfile.yaml`, logo, and outerloop resources. These devfile stacks give users a template to get started developing with devfiles. For more information on the structure of a devfile registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].
```

- You might need to get approval to publish a link to a GitHub repo. 
- Use `link:` with external links.

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:11]:
```suggestion
A devfile registry has two components: a devfile index server and an OCI registry server.
```

Comment (rolfedh) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:11]: IBM SG: "Use a colon after an independent clause to introduce an inline list."

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:15]: IBM SG: "**following** adjective, noun
To introduce a list, procedure, or subsequent items, always use "following" as an adjective before a noun. For example, do not write "Complete the following" or "The functions include the following". Instead, write
"Complete the following steps" or "The adapter has the following functions"."

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:19]:
```suggestion
* Bootstraps the OCI registry with the devfile stacks.
* Provides an API to interact with the OCI registry and to retrieve devfile stacks.
```

Comment (rolfedh) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:19]: IBM SG: **Punctuation in lists**
Write lists so that all or none of the items start with complete sentences. If list items comprise only complete sentences, include a period after each sentence. Except in definition lists, do not include any end punctuation if list items comprise only sentence fragments.
If a list contains list items that start with sentence fragments, and one or more of the fragments are followed by complete sentences, include periods after all sentence fragments and sentences.

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:21]: Use `link:`

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:25]:
```suggestion
The devfile stacks in a devfile registry are stored in an OCI-compatible artifact registry. The artifact registry is based on this link:https://github.com/distribution/distribution[reference implementation of an OCI registry].
```
- Broke this up to make it easier to read. To clarify _what_ is based on the reference implementation.
- You might have problems getting permission to link to GitHub repos. 
- Use `link:`

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:31]:
```suggestion
As an <role>, you can use multiple types of devfile registries to access devfile stacks and tools.
```
- IBM SG: Write clearly and directly. Wherever possible, use the second person ("you"). 
- If it helps, you can also identify the role of the user. For example "As an administrator, you..."

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:41]: Review content for multiple instances of "the user" or "a user's" that you can replace with "you" and "your."

File: docs/modules/user-guide/nav.adoc
Comment (johnmcollier) [reply docs/modules/user-guide/nav.adoc:2]: Yeah, that seems reasonable to me 👍

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (johnmcollier) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:33]: These are just showing up as plain text in the rendered doc rather than a subheader. Should they be subheaders?

**Review by johnmcollier — COMMENTED** [review]
Generally looks fine to me. Just left a couple small comments

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: Good call! Made the change, thanks!

Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: I was hoping you'd say that because that's my thought exactly! In the new commit, I'll remove the links, thanks!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:15]: Good revision. Made the change, thanks!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:31]: They look like subheaders to me. The screenshot I've included is what I see on my local server. Does this match with what you see?

Screen Shot 2021-06-07 at 3 46 13 PM

Comment (johnmcollier) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:31]: Okay that looks fine to me, must be a quirk with how GitHub renders the pages. So long as they look good on the website, that's good 👍

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc]: I removed the link, so it's no longer a concern, thanks. But it's good to know!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc]: Thanks, Rolfe, I've made the change, let me know what you think, thanks!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc]: Thanks, Rolfe, I've removed all the links. There's no only one xref under `additional resources`. I'll know to do this next time though when I have external links to include!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc]: Thanks, made the change!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:29]: Thanks for the suggestion! Made the change, thanks!

**Comment by jc-berger** [comment]
@johnmcollier and @rolfedh if you could both please take another look and give it the LGTM if either of you thinks it looks good. 

Apologies for the many commits...I accepted Rolfe's changes than made several myself, resulting in lots of commits! Thanks for the help!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:5]:
```suggestion
A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che, and the OpenShift Developer Console. Therefore, you can access devfiles through the devfile registry.  Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification, and devfile stacks are stored as OCI-artifacts in the registry.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: Is "outerloop" Red Hat jargon or is it a devfile term that is well-understood users?

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:33]:
```suggestion
The public community registry provides stacks for various community tools to consume, such as NodeJS, Quarkus, and Open Liberty.
```
IBM SG: "Use a colon after an _independent clause_ to introduce an inline list."

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: Good point. @johnmcollier, I'm thinking users will be unfamiliar with `logo` and `outerloop resources`; therefore, maybe we should remove them. What do you think?

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rolfedh) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]:
```suggestion
Each devfile stack corresponds to a specific runtime or framework, such as NodeJS, Quarkus, or Wildfly. A devfile stack also contains resources like a `devfile.yaml` file, logo, and outerloop resources. These devfile stacks give developers a template to get started developing cloud-native applications.
```

**Review by rolfedh — APPROVED** [review]
One minor comment. Otherwise, /lgtm

**Review by johnmcollier — APPROVED** [review]

**Comment by jc-berger** [comment]
@rkratky gotten the LTM from docs and devs. Please let me know if there's anything you would like me to address or change before you merge, thanks!

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:7]: @jc-berger,

* Always check project/product naming.
* "Outerloop" needs an explanation.

```suggestion
Each devfile stack corresponds to a specific runtime or framework, such as Node.js, Quarkus, or WildFly. A devfile stack also contains resources like a `devfile.yaml` file, logo, and outerloop resources. These devfile stacks provide developers with templates to get started developing cloud-native applications.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:1]: * Don't add the module type to the ID (regardless of what the current templates show).
* This is not a procedure module. The file needs to be renamed, and the Antora page fixed.

```suggestion
[id="devfile-registry_{context}"]
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:11]:
```suggestion
A devfile registry has two components:

 * Devfile index server
 * OCI registry server
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:17]:
```suggestion
* Uses the `index.json` file to obtain metadata about the stacks and samples and hosts this metadata in the registry for tools to consume.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:33]:
```suggestion
The public community registry provides stacks for various community tools to consume, such as Node.js, Quarkus, and Open Liberty.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:35]:
```suggestion
Locate the source for the stacks in the public community registry in the link:https://github.com/devfile/registry[devfile/registry repository].
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:39]:
```suggestion
Devfile registries can be operated on private Kubernetes clusters, deployed using either the devfile registry Operator or the devfile registry Helm Chart.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:41]:
```suggestion
The on-cluster devfile registries can be configured with custom devfile stacks suited to the needs of the owner organization.
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:45]:
```suggestion
* xref:authoring-stacks.adoc[].
```

File: docs/modules/user-guide/partials/proc_devfile-registry.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_devfile-registry.adoc:29]:
```suggestion
There are multiple types of devfile registries to access devfile stacks and tools.
```

**Review by rkratky — CHANGES_REQUESTED** [review]
@jc-berger, comments added.

File: docs/modules/user-guide/partials/con_devfile-registry.adoc
Comment (jc-berger) [inline docs/modules/user-guide/partials/con_devfile-registry.adoc:28]: From previous issues, I know we were concerned with inappropriately calling our users directly. But we have a lot of instances of passive voice throughout this doc, which is not advised by the style guide. 

@rkratky how concerned do you think we should be about passive voice in this doc before we merge? 

If we are fine with using second person, I can quickly make changes to active voice throughout. Like in this instance, I could write, "When you deploy a devfile registry, the stacks are pushed..." 

Otherwise, we can say something like, "When users deploy a devfile registry, the stacks are pushed..." 

Basically, I wanted to get your opinion before making so many changes to the verb tense throughout, thanks!

**Comment by jc-berger** [comment]
@rkratky, addressed your comment and made some more changes in my new commit. Also, you'll see a comment I made pointing out the many uses of passive voice. 

If you think it all looks fine, please go ahead and merge. Otherwise, I'm happy to continue to make changes. Thanks!

File: docs/modules/user-guide/partials/con_devfile-registry.adoc
Comment (rkratky) [reply docs/modules/user-guide/partials/con_devfile-registry.adoc:28]: @jc-berger, I don't have a problem with passive voice here. It's a concept doc, and it only provides a background explanation. It's not a procedure that would instruct users to do things. Forcing it wouldn't be helpful. E.g.:

_"When users deploy a devfile registry, the stacks are pushed..."_

That would be weird. It's usually not _users_ who deploy the registry. It's the admin, or maybe the service provider. We want the docs to be authoritative (_do this, do that_ instead of _consider blah blah_), but only when makes sense.

**Review by rkratky — APPROVED** [review]

**Pull request merged**
